### PR TITLE
Feat/config events

### PR DESCRIPTION
**Pull request by LordCheta**
This PR adds support for events to the PartnerConfiguration contract

https://rsklabs.atlassian.net/browse/RNS-183?atlOrigin=eyJpIjoiYTAzY2JmNGU0NzNkNGNlYjkyZDc2NjNkZmEzZjgzMDAiLCJwIjoiaiJ9

### CODE REVIEW

File: test/PartnerConfiguration.spec.ts [hunk 1]
```diff
@@ -343,4 +343,65 @@ describe('Partner Configuration', () => {
...
+
+    it('Should emit the MinCommitmentAgeChanged event with the correct params', async () => {
+      const NEW_MIN_COMMITMENT_AGE = DEFAULT_MIN_COMMITMENT_AGE + 1;
+      await expect(
+        PartnerConfiguration.setMinCommitmentAge(NEW_MIN_COMMITMENT_AGE)
+      )
+        .to.emit(PartnerConfiguration, 'MinCommitmentAgeChanged')
+        .withArgs(DEFAULT_MIN_COMMITMENT_AGE, NEW_MIN_COMMITMENT_AGE);
+    });
+  });
```
Comment (sergio-jara) [inline test/PartnerConfiguration.spec.ts:406]: kind of a pain in the ass but... can we move the events strings to a single location?

**Review by sergio-jara — CHANGES_REQUESTED** [review]
Looks good. 
One thing tho. I noticed it only added events for the PartnerConfiguration contract. The security report also mentioned 
PartnerRegistrar
FeeManager
PartnerManager.
do you plan to create a new PR for those?

File: contracts/PartnerConfiguration/IPartnerConfiguration.sol [hunk 2]
```diff
@@ -7,6 +7,62 @@ pragma solidity ^0.8.16;
  * @dev Defines the configuration for a partner in particular
  */
 interface IPartnerConfiguration {
+    /**
+     * @notice event emiited when the minimum length is changed
```
[the remark this answers is not in the document]
Comment (LordCheta) [reply contracts/PartnerConfiguration/IPartnerConfiguration.sol:11]: 🫡 thanks for that suggestion

File: test/PartnerConfiguration.spec.ts (hunk 1 above)
Comment (LordCheta) [reply test/PartnerConfiguration.spec.ts:406]: Sure thing

**Comment by LordCheta** [comment]
> Looks good. One thing tho. I noticed it only added events for the PartnerConfiguration contract. The security report also mentioned PartnerRegistrar FeeManager PartnerManager. do you plan to create a new PR for those?

Will address the other ones in a new PR so this doesn't get bigger.

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 3]
```diff
@@ -78,6 +78,14 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
        @inheritdoc IPartnerConfiguration
      */
     function setMinLength(uint256 minLength) external override onlyOwner {
+        uint256 preModifiedValue = _minLength;
+
+        if (preModifiedValue == minLength) {
+            revert(
+                "PartnerConfiguration: Param being modified is same as new param"
+            );
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:86]: please remove "PartnerConfiguration: ". the message needs to be shorter

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 4]
```diff
@@ -103,6 +113,14 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
        @inheritdoc IPartnerConfiguration
      */
     function setMaxLength(uint256 maxLength) external override onlyOwner {
+        uint256 preModifiedValue = _maxLength;
+
+        if (preModifiedValue == maxLength) {
+            revert(
+                "PartnerConfiguration: Param being modified is same as new param"
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:120]: same

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 5]
```diff
@@ -90,6 +98,8 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
         }
 
         _minLength = minLength;
+
+        emit MinLengthChanged(preModifiedValue, _minLength);
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:102]: I talked to Ulas about events being fires before or after and he said it is not important. But if you did it at the begiining in the other PR, just follow the same principle here

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 6]
```diff
@@ -163,6 +203,14 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
        @inheritdoc IPartnerConfiguration
      */
     function setMaxDuration(uint256 maxDuration) external override onlyOwner {
+        uint256 preModifiedValue = _maxDuration;
+
+        if (preModifiedValue == maxDuration) {
+            revert(
+                "PartnerConfiguration: Param being modified is same as new param"
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:210]: same

File: test/PartnerConfiguration.spec.ts [hunk 7]
```diff
@@ -18,6 +18,15 @@ const DEFAULT_DISCOUNT = 4;
 const DEFAULT_IS_UNICODE_SUPPORTED = false;
 const DEFAULT_FEE_PERCENTAGE = 5;
 
+const MIN_DURATION_CHANGED_EVENT = 'MinDurationChanged';
```
Comment (sergio-jara) [inline test/PartnerConfiguration.spec.ts:21]: nice. It could be in a separate file tho. in order to allow its usage outside this file.

**Review by sergio-jara — CHANGES_REQUESTED** [review]

**Review by sergio-jara — APPROVED** [review]

**Pull request merged**